src/historywork/ApplyLedgerChainWork.cpp

```diff
@@ -132,13 +132,15 @@ ApplyLedgerChainWork::applyHistoryOfSingleLedger()
     {
         if (hHeader.hash != lclHeader.header.previousLedgerHash)
         {
-            throw std::runtime_error(
+            auto msg =
                 fmt::format("replay of {:s} failed to connect on hash of LCL "
                             "predecessor {:s}",
                             LedgerManager::ledgerAbbrev(hHeader),
                             LedgerManager::ledgerAbbrev(
                                 lclHeader.header.ledgerSeq - 1,
-                                lclHeader.header.previousLedgerHash)));
+                                lclHeader.header.previousLedgerHash));
+            CLOG(ERROR, "History") << msg;
```

Comment:
Anti-pattern? Leave it to some handler, don't log twice. Pros: file and line number of the error. Maybe we can use some mechanism to contain line numbers in exceptions.